DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+```java
+public void drive(double lInput, double rInput) {
+    var lSpeed = kinematics.toWheelSpeeds(
+        new ChassisSpeeds(-lInput * MAX_SPEED, 0.0, 0));
+    var rSpeed = kinematics.toWheelSpeeds(
+        new ChassisSpeeds(-rInput * MAX_SPEED, 0.0, 0));
+    setSpeeds(lSpeed, rSpeed);
+}
+```
+We convert the linear input into wheel speeds for the left and right sides using the robot’s kinematics. Next we pass these wheel speeds to the `setSpeeds` method we just discussed. This method handles the actual motor control. The inputs are going to be the driver inputs from the controller. 
+
+### Unit Testing & System Checks
+#### Unit Tests
+
+Think of unit tests as a way to check that each small piece of your code—like a single motor or a sensor—is functioning properly on its own before you start putting them all together. It's much easier to fix problems when they’re isolated to a single component than when they’re all tangled up in the full system.
+##### MAKE SURE TO CHECK OUT THIS [GUIDE](https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/unit-testing.html) FROM WPILIB ON HOW TO SETUP UNIT TESTS.
+
+Let’s talk about the tests for our drivetrain. We'll start it by creating a driveTest.java file in the robot folder of the tests folder. 
+
+First, we need to set things up before we run any tests. This is done in the setup() method:
+```java
+Drive drive;
+CANSparkMax spark;
+
+  @BeforeEach
+  public void setup() {
+    setupTests();
+    drive = new Drive();
+    spark = drive.getSpark();
+  }
+```
+- ```getSpark()``` is a simple method in drive.java that lets us use one of the motors: 
+    - ``` public CANSparkMax getSpark(){return leftLeader;}```
+
+Next, we will need to reset our drive between every tests so we are able to experiment different things at once. 
+```java
+  @AfterEach
+  void destroy() throws Exception {
+    reset(drive);
+  }
+```
+
+Once everything is set up, you can make tests with the `@Test` annotation. Here's a quick example of testing if our motors are getting voltage:
+```java
+   @Test
+  public void testSparks() {
+    drive.drive(1, 0);
+
+    assert spark.getBusVoltage() > 3;
+  }
+  ```
+  - 3 is a random number here, but it is small on purpose. The motor will need more voltage than 3 in order to properly run.
+
+  Another example: Here we are testing to see if our pose is changing as we drive: 
+```java
+  @Test
+  public void testPose(){
+    drive.drive(0.8, 0.8);
+    fastForward();
+
+    assertNotEquals(new Pose2d(), drive.getPose());
+  }
+```
+  #### Running the tests
+To actually see if the tests are valid, open the wpilib command palette(ctrl + shift + p) and search ```Test Robot Code``` and press enter. After a bit, it will show you the results of all your tests in the terminal. 
```

Comment:
```suggestion
To actually see if the tests are valid, open the WPILib command palette (top right corner logo) and search `Test Robot Code` and press enter. After it runs, it will show you the results of all your tests in the terminal. 
```
More accessible to access for different platforms